core/src/main/java/org/mule/runtime/core/privileged/processor/MessageProcessors.java

```diff
@@ -99,6 +104,35 @@ public static MessageProcessorChain newChain(Optional<ProcessingStrategy> proces
     }
   }
 
+  public static MessageProcessorChain newChain(Optional<ProcessingStrategy> processingStrategy, List<Processor> processors,
+                                               String name) {
+    if (processors.size() == 1 && processors.get(0) instanceof MessageProcessorChain) {
+      return (MessageProcessorChain) processors.get(0);
+    } else {
+      return buildNewChainWithListOfProcessors(processingStrategy, processors, new SpanCustomizationInfo() {
```

Comment:
Verify if this can be set in a different class that implements SpanCustomizationInfo.